### ISSUE TITLE
Auto save behaviour might be unexpected here

### ISSUE THREAD

**Issue by sbatten** [microsoft/vscode#141473]
Testing #141347

I have auto save enabled. This makes sense when editing files, but feels different when editing memory during debugging.

**Comment by connor4312** [comment]
@lramos15 do you know if there's a way to disable autosave on a per-editor basis?

**Comment by lramos15** [comment]
There is currently none that I'm aware of. @bpasero would know more. Maybe we want to explore more granular configuration for that. Although what's the issue with autosave here exactly is it that if you're editing memory and it saves prematurely it may mess things up?

**Comment by bpasero** [comment]
No, see https://github.com/microsoft/vscode/issues/42170. It is not so trivial to do per editor/folder but maybe I could revisit that.

**Comment by connor4312** [comment]
This is also something I wished I had working on the [onedrive browser prototype](https://github.com/microsoft/vscode-onedrive-browser) (which is likely to be published eventually) since saving files to OneDrive is not very fast and creates "revisions" which can be confusing if numerous.

**Comment by JamesDougherty** [comment]
@bpasero Please do as this would be very beneficial. An example use case, and the reason I searched for this, is I currently have several VS Code instances open. One for an Angular project that drives our frontend and one for our AWS/API that drives our backend. I want the autosave feature enabled for the AWS/API project because I don't want to miss any changes during a deployment (you get used to the autosave feature and forgot to save at times). However, the Angular project is pretty large and it takes a while to rebuild when using the live development server. If the autosave feature is on, then it will constantly keep trying to rebuild on every autosave. So, I leave the autosave feature off and when I'm ready to test changes I just use the save all (CTRL+K S) shortcut keys. Then it only does it once. I have to remember that if I turn off the autosave on the Angular project, then it turns it off for the AWS/API project as well. So, enabling and disabling the autosave feature affects all instances of VS Code open.

**Comment by bpasero** [comment]
@connor4312 auto save can be configured per resource and/or editor since a few milestones and we use this already to disable auto save when AI suggestions are made. See along this method:

https://github.com/microsoft/vscode/blob/676cbd90fb60759b2262638e1e06121fb7cc36c4/src/vs/workbench/services/filesConfiguration/common/filesConfigurationService.ts#L95
